reader-osm/src/test/java/com/graphhopper/GraphHopperIT.java

```diff
@@ -1356,6 +1356,54 @@ public void testFlexMode_631() {
         // note: combining hybrid & speed mode is currently not possible and should be avoided: #1082
     }
 
+    @Test
+    public void testCrossQuery() {
+        final String profile1 = "p1";
+        final String profile2 = "p2";
+        final String profile3 = "p3";
+        final String vehicle = "car";
+        GraphHopper hopper = createGraphHopper(vehicle).
+                setOSMFile(MONACO).
+                setProfiles(
+                        new ProfileConfig(profile1).setVehicle("car").setWeighting("short_fastest").putHint("short_fastest.distance_factor", 0.07),
+                        new ProfileConfig(profile2).setVehicle("car").setWeighting("short_fastest").putHint("short_fastest.distance_factor", 0.10),
+                        new ProfileConfig(profile3).setVehicle("car").setWeighting("short_fastest").putHint("short_fastest.distance_factor", 0.15)
+                ).
```

Comment:
This is a bit artificial, because we can also just use profile1 and change the distance factor on every request, but creating a dedicated profile for this illustrates the new feature here: We can setup a profile on the server side and then do no longer have to specify the different parameters (like the distance factor) and at the same time we do not need an extra landmark calculation for this. This should be useful for #1776 when we want to setup a complex profile and then just select it by using the `profile` parameter (which btw is still missing on the web layer (#1859)).